src/utils/typedData.ts

```diff
@@ -20,18 +72,13 @@ function getHex(value: BigNumberish): string {
 
 /**
  * Validates that `data` matches the EIP-712 JSON schema.
- *
- * @param {any} data
- * @return {boolean}
  */
-const validateTypedData = (data: unknown): data is TypedData => {
+function validateTypedData(data: unknown): data is TypedData {
```

Comment:
it would be nice to have some extra checks like:
- no types with empty name
- no types ending in `*`
- not types like in parenthesis
- not types that try to override the presets
- no duplicated types
- all enum variants types must be enclosed in parenthesis
- no unused types (types that are not being used by any other type)

Reply:
More robust validation will definitely be added. The only question is whether it will be within this PR or a separate one.